src/operators/infinitepepo.jl

```diff
@@ -3,13 +3,14 @@
 
 Represents an infinite projected entangled-pair operator (PEPO) on a 3D cubic lattice.
 """
-struct InfinitePEPO{T<:PEPOTensor} <: InfiniteSquareNetwork{T,3}
+struct InfinitePEPO{T<:PEPOTensor} <: InfiniteGridNetwork{T,3}
     A::Array{T,3}
 
     function InfinitePEPO(A::Array{T,3}) where {T<:PEPOTensor}
         # space checks
         for (d, w, h) in Tuple.(CartesianIndices(A))
-            space(A[d, w, h], 1) == space(A[d, w, _next(h, end)], 2)' ||
+            only(codomain(physicalspace(A[d, w, h]))) ==
+            only(domain(physicalspace(A[d, w, _next(h, end)]))) ||
```

Comment:
This looks strange to me though, I would expect `physicalspace` of an operator to return the space it is acting on, not the actual homspace/product space mapping it

Reply:
How about I use `domain_physicalspace` and `codomain_physicalspace` here? Just having `physicalspace` return a single space is not really sensible I think, since it often happens that the physical domain and codomain are different for symmetric multilayer PEPOs.

Reply:
I updated the space checks to explicitly use `domain_physicalspace` and `codomain_physicalspace`. That should solve the issue here. Resolving this and moving the discussion on what to do with `physicalspace` further down.